lib/metanorma/collection_renderer.rb

```diff
@@ -0,0 +1,267 @@
+    def coverpage
+      File.open(File.join(@outdir, "index.html"), "w:UTF-8") do |f|
+        f.write  @isodoc.populate_template(File.read(@coverpage))
+      end
+    end
+
+    def indexfile_title(m)
+      lvl = m&.at(ns("./level"))&.text&.capitalize
+      lbl = m&.at(ns("./title"))&.text
+      "#{lvl}#{ lvl && lbl ? ": ": "" }#{lbl}"
+    end
+
+    # uses the identifier to label documents; other attributes (title) can be looked up
+    # in @files[id][:bibdata]
+    def indexfile_docref(m)
+      return "" unless m.at(ns("./docref"))
+      ret = "<ul>\n"
+      m.xpath(ns("./docref")).each do |d|
+        identifier = d.at(ns("./identifier")).text
+        link = d["fileref"] ? d["fileref"].sub(/\.xml$/, ".html") : d["id"] + ".html"
+        ret += "<li><a href='./#{link}'>#{identifier}</a></li>\n"
+      end
+      ret += "</ul>\n"
+      ret
+    end
+
+    # single level navigation list, with hierarchical nesting
+    # if multiple lists are needed as separate HTML fragments, multiple instances of this function will be needed,
+    # and associated to different variables in the call to @isodoc.metadata_init (including possibly an array of HTML fragments)
+    def indexfile(m)
+      ret = "<ul>\n"
+      ret += "<li>#{indexfile_title(m)}</li>\n"
+      ret += indexfile_docref(m)
+      m.xpath(ns("./manifest")).each do |d|
+        ret += "#{indexfile(d)}\n"
+      end
+      ret += "</ul>\n"
+      ret
+    end
+
+    # return file contents + output filename for each file in the collection, given a docref entry
+    # @param data [Hash]
+    # @param read [Boolean]
+    # @return [Array<String, nil>]
+    def targetfile(data, read = false)
+      if data[:type] == "fileref" then ref_file data[:ref], read
+      else xml_file data[:id], read
+      end
+    end
+
+    # @param ref [String]
+    # @param read [Boolean]
+    # @return [Array<String, nil>]
+    def ref_file(ref, read)
+      file = File.read(ref, encoding: "utf-8") if read
+      filename = ref.sub(/\.xml$/, ".html")
+      [file, filename]
+    end
+
+    # @param id [String]
+    # @param read [Boolean]
+    # @return [Array<String, nil>]
+    def xml_file(id, read)
+      file = @xml.at(ns("//doc-container[@id = '#{id}']")).to_xml if read
+      filename = id + ".html"
+      [file, filename]
+    end
+
+    def update_bibitem(b, docid, identifier)
+      docid = b&.at(ns("./docidentifier"))&.text
+      unless @files[docid]
+        warn "Cannot find crossreference to document #{docid} in document #{identifier}!"
+        abort
+      end
+      id = b["id"]
+      newbib = b.replace(@files[docid][:bibdata])
+      newbib.name = "bibitem"
+      newbib["id"] = id
+      newbib&.at(ns("./ext"))&.remove
+      file, url = targetfile(@files[docid], false)
+      newbib.at(ns("./docidentifier")).previous = %{<uri type="citation">#{url}</uri>}
+    end
+
+    def repo_docid(docid)
+      docid.sub(%r{^current-metanorma-collection/}, "")
```

Comment:
What does this do?

Reply:
> What does this do?

The rendered made by @opoudjis, I'm just integrating it into metanorma. We should ask him what the code does.

Reply:
I think we can remove the code

Reply:
Ping @opoudjis for an answer.